client/client.go

```diff
@@ -18,6 +18,7 @@ type VanClient struct {
 	KubeClient  kubernetes.Interface
 	RouteClient *routev1client.RouteV1Client
 	RestConfig  *restclient.Config
+	SecretFile  string
```

Comment:
I see your point. But I have a few concerns. 
- we might have multiple tokens, so they would need to be stored in a slice
- if we store then, they will only be available if token were generated by the active pod (if pod was recycled, you'd lose that info)

Maybe instead of storing the secret file name, you could compare the value of the 
metadata.annotations.inter-router-host and inter-router-port from the provided secret
file. If those two match to the running host, then you proceed with rejecting the "self-connection".

Reply:
Thanks -- that sounds more reasonable -- I will try it.